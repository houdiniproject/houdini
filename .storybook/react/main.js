// License: LGPL-3.0-or-later
// Based on https://github.com/rails/webpacker/issues/1004#issuecomment-628377930
process.env.NODE_ENV = "development";
const railsWebpackEnv = require("../../config/webpack/environment");

module.exports = {
  stories: [
    "../../app/packs/**/*!(--html).stories.mdx",
    "../../app/packs/**/*!(--html).stories.[tj]s?(x)"
  ],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    'storybook-addon-intl'
  ],
  webpackFinal: (config) => {
    const result = {
    // do mutation to the config
    ...config,
    resolve: {
      ...config.resolve,
      ...railsWebpackEnv.config.resolve,
      modules: railsWebpackEnv.resolvedModules.map((i) => i.value),
    },
    module: {
      ...config.module,
      rules: railsWebpackEnv.loaders
        .filter((i) => !["nodeModules", //not sure why this is here
        "moduleCss" // this addresses issues with our webpack config for css not matching what storybook wants
      ].includes(i.key) )
        .map((i) => i.value),
    },
    plugins: [
      ...config.plugins,
      ...railsWebpackEnv.plugins.map((i) => i.value),
    ],
    
  };

  result.module.rules.filter((i) => i.test.test('.ttf')).forEach((i) => {
    i.use=['url-loader']
  })


  return result;
},
};
