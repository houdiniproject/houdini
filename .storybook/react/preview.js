// License: LGPL-3.0-or-later
import intlDecorate from '../../app/packs/components/tests/intl';
const jest = require('jest-mock');
window.jest = jest;

export const decorators = [intlDecorate()]