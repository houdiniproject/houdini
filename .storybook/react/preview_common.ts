// License: LGPL-3.0-or-or-later

import intlDecorate from '../../app/javascript/components/tests/intl';
import cssBaseline from '../../app/javascript/components/tests/decorators/baseline'
import clearSessionStorage from '../../app/javascript/components/tests/decorators/clearSessionStorage'
import { initialize, mswDecorator } from 'msw-storybook-addon';

initialize({
	onUnhandledRequest: 'bypass' // we need this because otherwise HMR and other parts of Storybook loading wouldn't work
});


export const decorators = [ clearSessionStorage(), mswDecorator, intlDecorate(), cssBaseline()]