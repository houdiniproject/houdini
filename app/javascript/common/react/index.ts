// License: LGPL-3.0-or-later

/**
 * Adds type safety to your default props.
 *
 * @example
 * // your component
 * function CustomComponent(props:{className:string, disabled?:boolean}) : JSX.Element {
 * 	// not show for
 * }
 *
 * CustomComponent.defaultProps = defaultProps(CustomComponent, {
 * 	className: 'another'
 * })
 *
 * @example <caption>Function properly typechecks</caption>
 * // use the same component as previous example
 * CustomComponent.defaultProps = defaultProps(CustomComponent, {
 * 	className: 'another',
 * 	unexpectedProp: 3 // <-- Typecheck will throw a compile error because you didn't expect this.
 * })
 *
 * @param {(props:TProperty) => JSX.Element} _component - the component you want to create default props for
 * @param {Partial<TProperty>} defaultProps - the defaultProps for your component
 * @returns defaultProps
 * */
export function defaultProps<TProperty>(_component:(props:TProperty) => JSX.Element, defaultProps:Partial<TProperty>):Partial<TProperty> {
	return defaultProps;
}