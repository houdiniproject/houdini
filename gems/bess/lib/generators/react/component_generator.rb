# frozen_string_literal: true

# License: AGPL-3.0-or-later WITH WTO-AP-3.0-or-later
# Full license explanation at https://github.com/houdiniproject/houdini/blob/master/LICENSE
# from: https://github.com/reactjs/react-rails/blob/master/lib/generators/react/component_generator.rb
module React
    module Generators
      class ComponentGenerator < ::Rails::Generators::NamedBase
        source_root File.expand_path '../../templates', __FILE__
        desc <<-DESC.strip_heredoc
        Description:
            Scaffold a React component into `components/` of your Webpacker source or asset pipeline.
            The generated component will include a basic render function and a PropTypes
            hash to help with development.
        Available field types:
            Basic prop types do not take any additional arguments. If you do not specify
            a prop type, the generic node will be used. The basic types available are:
            any
            array
            bool
            element
            func
            number
            object
            node
            shape
            string
            Special PropTypes take additional arguments in {}, and must be enclosed in
            single quotes to keep bash from expanding the arguments in {}.
            instanceOf
            takes an optional class name in the form of {className}
            oneOf
            behaves like an enum, and takes an optional list of strings that will
            be allowed in the form of 'name:oneOf{one,two,three}'.
            oneOfType.
            oneOfType takes an optional list of react and custom types in the form of
            'model:oneOfType{string,number,OtherType}'
        Examples:
            rails g react:component person name
            rails g react:component restaurant name:string rating:number owner:instanceOf{Person}
            rails g react:component food 'kind:oneOf{meat,cheese,vegetable}'
            rails g react:component events 'location:oneOfType{string,Restaurant}'
        DESC
  
        argument :attributes,
                 :type => :array,
                 :default => [],
                 :banner => 'field[:type] field[:type] ...'

        class_option :ts,
                     type: :boolean,
                     default: true,
                     desc: 'Output tsx class based component'

        REACT_PROP_TYPES = {
          'node' =>        'PropTypes.node',
          'bool' =>        'PropTypes.bool',
          'boolean' =>     'PropTypes.bool',
          'string' =>      'PropTypes.string',
          'number' =>      'PropTypes.number',
          'object' =>      'PropTypes.object',
          'array' =>       'PropTypes.array',
          'shape' =>       'PropTypes.shape({})',
          'element' =>     'PropTypes.element',
          'func' =>        'PropTypes.func',
          'function' =>    'PropTypes.func',
          'any' =>         'PropTypes.any',
  
          'instanceOf' => ->(type) {
            'PropTypes.instanceOf(%s)' % type.to_s.camelize
          },
  
          'oneOf' => ->(*options) {
            enums = options.map{ |k| "'#{k.to_s}'" }.join(',')
            'PropTypes.oneOf([%s])' % enums
          },
  
          'oneOfType' => ->(*options) {
            types = options.map{ |k| "#{lookup(k.to_s, k.to_s)}" }.join(',')
            'PropTypes.oneOfType([%s])' % types
          }
        }
  
        TYPESCRIPT_TYPES = {
          'node' =>        'React.ReactNode',
          'bool' =>        'boolean',
          'boolean' =>     'boolean',
          'string' =>      'string',
          'number' =>      'number',
          'object' =>      'object',
          'array' =>       'Array<any>',
          'shape' =>       'object',
          'element' =>     'object',
          'func' =>        'object',
          'function' =>    'object',
          'any' =>         'any',
  
          'instanceOf' => ->(type) {
            type.to_s.camelize
          },
  
          'oneOf' => ->(*opts) {
            opts.map{ |k| "'#{k.to_s}'" }.join(" | ")
          },
  
          'oneOfType' => ->(*opts) {
            opts.map{ |k| "#{ts_lookup(k.to_s, k.to_s)}" }.join(" | ")
          }
        }
  
        def create_component_file
          template_extension = if options[:coffee]
            'js.jsx.coffee'
          elsif options[:ts]
            'js.jsx.tsx.erb'
          elsif options[:es6] || webpacker?
            'es6.jsx'
          else
            'js.jsx'
          end
  
          # Prefer webpacker to sprockets:
          if webpacker?
            new_file_name = file_name.camelize
            extension = if options[:coffee]
              'coffee'
            elsif options[:ts]
              'tsx'
            else
              'js'
            end
            target_dir = webpack_configuration.source_path
              .join('components')
              .relative_path_from(::Rails.root)
              .to_s
          else
            new_file_name = file_name
            extension = template_extension
            target_dir = 'app/assets/javascripts/components'
          end
  
          file_path = File.join(target_dir, class_path, "#{new_file_name}.#{extension}")
          template("component.#{template_extension}", file_path)
        end
  
        private
  
        def webpack_configuration
          Webpacker.respond_to?(:config) ? Webpacker.config : Webpacker::Configuration
        end
  
        def component_name
          file_name.camelize
        end
  
        def file_header
          if webpacker?
            return %|import * as React from "react"\n| if options[:ts]
            %|import React from "react"\nimport PropTypes from "prop-types"\n|
          else
            ''
          end
        end
  
        def file_footer
          if webpacker?
            %|export default #{component_name}|
          else
            ''
          end
        end
  
        def webpacker?
          defined?(Webpacker)
        end
  
        def parse_attributes!
          self.attributes = (attributes || []).map do |attr|
            name = ''
            type = ''
            args = ''
            args_regex = /(?<args>{.*})/
  
            name, type = attr.split(':')
  
            if matchdata = args_regex.match(type)
              args = matchdata[:args]
              type = type.gsub(args_regex, '')
            end
  
            if options[:ts]
              { :name => name, :type => ts_lookup(name, type, args), :union => union?(args) }
            else
              { :name => name, :type => lookup(type, args) }
            end
          end
        end
  
        def union?(args = '')
          return args.to_s.gsub(/[{}]/, '').split(',').count > 1
        end
  
        def self.ts_lookup(name, type = 'node', args = '')
          ts_type = TYPESCRIPT_TYPES[type]
          if ts_type.blank?
            if type =~ /^[[:upper:]]/
              ts_type = TYPESCRIPT_TYPES['instanceOf']
            else
              ts_type = TYPESCRIPT_TYPES['node']
            end
          end
  
          args = args.to_s.gsub(/[{}]/, '').split(',')
  
          if ts_type.respond_to? :call
            if args.blank?
              return ts_type.call(type)
            end
  
            ts_type = ts_type.call(*args)
          end
  
          ts_type
        end
  
        def ts_lookup(name, type = 'node', args = '')
          self.class.ts_lookup(name, type, args)
        end
  
         def self.lookup(type = 'node', options = '')
           react_prop_type = REACT_PROP_TYPES[type]
           if react_prop_type.blank?
             if type =~ /^[[:upper:]]/
               react_prop_type = REACT_PROP_TYPES['instanceOf']
             else
               react_prop_type = REACT_PROP_TYPES['node']
             end
           end
  
           options = options.to_s.gsub(/[{}]/, '').split(',')
  
           react_prop_type = react_prop_type.call(*options) if react_prop_type.respond_to? :call
           react_prop_type
         end
  
         def lookup(type = 'node', options = '')
           self.class.lookup(type, options)
         end
      end
    end
  end