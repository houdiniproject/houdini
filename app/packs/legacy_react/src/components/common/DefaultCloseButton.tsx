import * as React from "react";
import { action, observable } from "mobx";
import { Transition } from "react-transition-group";
import { CloseButton } from "./svg/CloseButton";
import color from "color";
import { observer } from "mobx-react";
import ScreenReaderOnlyText from "./ScreenReaderOnlyText";

interface DefaultCloseButtonProps {
  onClick?: () => void
}

const mainColor = '#969696'
const darkenedColor = color(mainColor).darken(0.1).hex()
const defaultStyles = {
  foreground: {
    fill: mainColor
  },
  background: {
    stroke: mainColor,
    fill: '#FFFFFF'
  }
}

const states: { [state: string]: any } = {
  entering: {
    foreground: {
      fill: darkenedColor,
      transition: 'fill 250ms ease-in-out'
    },
    background: {
      stroke: darkenedColor,
      transition: 'stroke 250ms ease-in-out'
    }
  },
  entered: {
    foreground: {
      fill: darkenedColor,
    },
    background: {
      stroke: darkenedColor,
    }
  },
  exiting: {
    foreground: {
      fill: mainColor,
      transition: 'fill 250ms ease-in-out'
    },
    background: {
      stroke: mainColor,
      transition: 'stroke 250ms ease-in-out'
    }
  },
  exited: {
    foreground: {
      fill: mainColor,
    },
    background: {
      stroke: mainColor,
    }
  }
}

@observer
export class DefaultCloseButton extends React.Component<DefaultCloseButtonProps, {}> {
  @observable
  hovering: boolean

  @observable
  focusing: boolean

  @action.bound
  mouseEnter() {
    this.hovering = true;
  }

  @action.bound
  mouseLeave() {
    this.hovering = false;
  }

  @action.bound
  keyDown(event: React.KeyboardEvent<HTMLAnchorElement>) {
    if (event.key == 'Enter') {
      event.preventDefault();
      this.props.onClick();
    }
  }

  render() {
    return <Transition in={this.hovering} timeout={250}>
      {(hoverState) => {
        const backgroundStyle = {
          ...defaultStyles.background,
          ...((states[hoverState] && states[hoverState].background) || {})
        }

        const foregroundStyle =
        {
          ...defaultStyles.foreground,
          ...((states[hoverState] && states[hoverState].foreground) || {})
        }

        return <a onMouseEnter={this.mouseEnter} onMouseLeave={this.mouseLeave} onClick={this.props.onClick} onKeyDown={this.keyDown} tabIndex={0} className={'focusable_item'}>
          <CloseButton backgroundCircleStyle={backgroundStyle}
            foregroundCircleStyle={foregroundStyle}
          /><ScreenReaderOnlyText>Close modal</ScreenReaderOnlyText>
        </a>

      }
      }
    </Transition>

  }
}
