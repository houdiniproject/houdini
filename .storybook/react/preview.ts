// License: LGPL-3.0-or-later
import {decorators} from './preview_common';

require('../../app/javascript/components/tests/decorators/roboto')

export {decorators};
