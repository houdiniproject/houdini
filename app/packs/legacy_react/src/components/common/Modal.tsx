// License: LGPL-3.0-or-later
import * as React from 'react';
import { observer } from 'mobx-react';
import AriaModal from 'react-aria-modal';
import { DefaultCloseButton } from './DefaultCloseButton';
import BootstrapWrapper from './BootstrapWrapper';
import { Row, Column } from './layout';

export interface ModalProps {
  onClose?: () => void // if you want your modal to close, this needs to set modalActive to false
  modalActive?: boolean
  titleText?: string
  focusDialog?: boolean
  dialogStyle?: any
  showCloseButton?: boolean
  childGenerator: () => any
}

class Modal extends React.Component<ModalProps, {}> {

  static defaultProps = {
    dialogStyle: { minWidth: '768px' },
    showCloseButton: true
  }

  render() {
    const modal = this.props.modalActive ? <AriaModal mounted={this.props.modalActive} titleText={this.props.titleText} focusDialog={this.props.focusDialog}
      onExit={this.props.onClose} dialogStyle={this.props.dialogStyle}>
      <BootstrapWrapper>
        <header className='modal-header' style={{
          position: 'relative',
          padding: '12px 10px 12px 20px'
        }}>
          <Row>
            <Column colSpan={11} breakSize={'xs'}>
              <h3 className='modal-header-title' style={{ margin: 0 }}>{this.props.titleText}</h3>
            </Column>
            {this.props.showCloseButton ?
              <Column colSpan={1} breakSize={'xs'}>
                <div style={{ textAlign: 'right' }}>
                  <DefaultCloseButton onClick={() => this.props.onClose()} />
                </div>
              </Column> : false
            }
          </Row>
        </header>
        <div className="modal-body">
          <div style={{ position: 'relative' }}>
            {this.props.childGenerator()}
          </div>
        </div>
      </BootstrapWrapper>
    </AriaModal> : false

    return modal
  }

}

export default observer(Modal)



