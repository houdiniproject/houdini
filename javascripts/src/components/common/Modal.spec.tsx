// License: LGPL-3.0-or-later
import * as React from 'react';
import 'jest';
import Modal, {ModalProps} from './Modal'
import {shallow, mount, ReactWrapper} from "enzyme";
import toJson from "enzyme-to-json";
import { DefaultCloseButton } from './DefaultCloseButton';

describe('Modal', () => {
  test('nothing displayed if inactive', () => {
    let modal = shallow(<Modal childGenerator={() => <div/>}/>)

    expect(toJson(modal)).toMatchSnapshot()
  })

  describe('active modal displays', () => {
    let onCloseWasCalled = false
    let modal:ReactWrapper
    beforeEach(() => {
      onCloseWasCalled = false;
      modal = mount(<Modal titleText={"title text"}
      focusDialog={true}
      modalActive={true}
      onClose={() => { onCloseWasCalled = true}}
      childGenerator={() => <div/>}/>)
    })

    it('matches snapshot', () => {
      expect(toJson(modal)).toMatchSnapshot()
    })

    it('closes on modal component close', () => {
      let modalComponent = modal.instance() as React.Component<ModalProps, {}> //casting to modal didn't work for reasons?
      modalComponent.props.onClose()
      expect(onCloseWasCalled).toBeTruthy()
    })

    it('closes on closeButtonClick', () => {
      let closeButton = modal.find('DefaultCloseButton')
      let instanceButton = closeButton.instance() as DefaultCloseButton
      instanceButton.props.onClick();
      expect(onCloseWasCalled).toBeTruthy()
    })
  })


  it('doesnt have a close button if we ask for none', () => {
    let onCloseWasCalled = false
    let modal = mount(<Modal titleText={"title text"}
                               focusDialog={true}
                               modalActive={true}
                               showCloseButton={false}
                               onClose={() => { onCloseWasCalled = true}}
                               childGenerator={() => <div/>}
                                />)
    expect(modal.find('DefaultCloseButton').exists()).toBeFalsy()
  })
  
})