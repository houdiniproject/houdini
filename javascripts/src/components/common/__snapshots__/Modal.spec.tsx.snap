// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal active modal displays matches snapshot 1`] = `
<Modal
  childGenerator={[Function]}
  dialogStyle={
    Object {
      "minWidth": "768px",
    }
  }
  focusDialog={true}
  modalActive={true}
  onClose={[Function]}
  showCloseButton={true}
  titleText="title text"
>
  <Displaced
    dialogStyle={
      Object {
        "minWidth": "768px",
      }
    }
    focusDialog={true}
    mounted={true}
    onExit={[Function]}
    titleText="title text"
  >
    <Portal
      containerInfo={
        <div>
          <div>
            <div
              style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; z-index: 1050; overflow-x: hidden; overflow-y: auto; text-align: center; background: rgba(0, 0, 0, 0.5); cursor: pointer;"
            >
              <div
                aria-label="title text"
                id="react-aria-modal-dialog"
                role="dialog"
                style="display: inline-block; text-align: left; top: 0px; max-width: 100%; cursor: default; outline: 0; min-width: 768px;"
                tabindex="-1"
              >
                <div
                  class="tw-bs"
                >
                  <header
                    class="modal-header"
                    style="position: relative; padding: 12px 10px 12px 20px;"
                  >
                    <div
                      class="row"
                    >
                      <h3
                        class="col-xs-11 modal-header-title"
                        style="margin: 0px;"
                      >
                        title text
                      </h3>
                      <div
                        class="col-xs-1 "
                        style="text-align: right;"
                      >
                        <a
                          class="focusable_item"
                          tabindex="0"
                        >
                          <svg
                            height="24"
                            version="1.1"
                            viewBox="0 0 24 24"
                            width="24"
                          >
                            <circle
                              cx="12"
                              cy="12"
                              r="10.65625"
                              style="stroke: #969696; fill: #FFFFFF;"
                            />
                            <circle
                              cx="12"
                              cy="12"
                              r="9.25"
                              style="fill: #969696;"
                            />
                            <path
                              d="M 7.130438,16.869562 16.869562,7.130438"
                              style="stroke: #ffffff; stroke-width: 1.375; stroke-linecap: square;"
                            />
                            <path
                              d="M 16.869562,16.869562 7.130438,7.130438"
                              style="stroke: #ffffff; stroke-width: 1.375; stroke-linecap: square;"
                            />
                          </svg>
                          <span
                            style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); border: 0px;"
                          >
                            Close modal
                          </span>
                        </a>
                      </div>
                    </div>
                  </header>
                  <div
                    class="modal-body"
                  >
                    <div
                      style="position: relative;"
                    >
                      <div />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <Modal
        dialogId="react-aria-modal-dialog"
        dialogStyle={
          Object {
            "minWidth": "768px",
          }
        }
        escapeExits={true}
        focusDialog={true}
        focusTrapPaused={false}
        includeDefaultStyles={true}
        mounted={true}
        onExit={[Function]}
        scrollDisabled={true}
        titleText="title text"
        underlayClickExits={true}
        underlayColor="rgba(0,0,0,0.5)"
        underlayProps={Object {}}
      >
        <FocusTrap
          _createFocusTrap={[Function]}
          active={true}
          focusTrapOptions={
            Object {
              "escapeDeactivates": true,
              "initialFocus": "#react-aria-modal-dialog",
            }
          }
          paused={false}
          tag="div"
        >
          <div>
            <div
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "rgba(0,0,0,0.5)",
                  "cursor": "pointer",
                  "height": "100%",
                  "left": 0,
                  "overflowX": "hidden",
                  "overflowY": "auto",
                  "position": "fixed",
                  "textAlign": "center",
                  "top": 0,
                  "width": "100%",
                  "zIndex": 1050,
                }
              }
            >
              <div
                aria-label="title text"
                id="react-aria-modal-dialog"
                key="b"
                role="dialog"
                style={
                  Object {
                    "cursor": "default",
                    "display": "inline-block",
                    "maxWidth": "100%",
                    "minWidth": "768px",
                    "outline": 0,
                    "textAlign": "left",
                    "top": 0,
                  }
                }
                tabIndex="-1"
              >
                <BootstrapWrapper>
                  <div
                    className="tw-bs"
                  >
                    <header
                      className="modal-header"
                      style={
                        Object {
                          "padding": "12px 10px 12px 20px",
                          "position": "relative",
                        }
                      }
                    >
                      <Row>
                        <div
                          className="row"
                        >
                          <Column
                            breakSize="xs"
                            colSpan={11}
                          >
                            <h3
                              className="col-xs-11 modal-header-title"
                              style={
                                Object {
                                  "margin": 0,
                                }
                              }
                            >
                              title text
                            </h3>
                          </Column>
                          <Column
                            breakSize="xs"
                            colSpan={1}
                          >
                            <div
                              className="col-xs-1 "
                              style={
                                Object {
                                  "textAlign": "right",
                                }
                              }
                            >
                              <DefaultCloseButton
                                onClick={[Function]}
                              >
                                <Transition
                                  appear={false}
                                  enter={true}
                                  exit={true}
                                  in={false}
                                  mountOnEnter={false}
                                  onEnter={[Function]}
                                  onEntered={[Function]}
                                  onEntering={[Function]}
                                  onExit={[Function]}
                                  onExited={[Function]}
                                  onExiting={[Function]}
                                  timeout={250}
                                  unmountOnExit={false}
                                >
                                  <a
                                    className="focusable_item"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    tabIndex={0}
                                  >
                                    <Component
                                      backgroundCircleStyle={
                                        Object {
                                          "fill": "#FFFFFF",
                                          "stroke": "#969696",
                                        }
                                      }
                                      foregroundCircleStyle={
                                        Object {
                                          "fill": "#969696",
                                        }
                                      }
                                    >
                                      <svg
                                        height="24"
                                        version="1.1"
                                        viewBox="0 0 24 24"
                                        width="24"
                                      >
                                        <circle
                                          cx="12"
                                          cy="12"
                                          r="10.65625"
                                          style={
                                            Object {
                                              "fill": "#FFFFFF",
                                              "stroke": "#969696",
                                            }
                                          }
                                        />
                                        <circle
                                          cx="12"
                                          cy="12"
                                          r="9.25"
                                          style={
                                            Object {
                                              "fill": "#969696",
                                            }
                                          }
                                        />
                                        <path
                                          d="M 7.130438,16.869562 16.869562,7.130438"
                                          style={
                                            Object {
                                              "stroke": "#ffffff",
                                              "strokeLinecap": "square",
                                              "strokeWidth": 1.375,
                                            }
                                          }
                                        />
                                        <path
                                          d="M 16.869562,16.869562 7.130438,7.130438"
                                          style={
                                            Object {
                                              "stroke": "#ffffff",
                                              "strokeLinecap": "square",
                                              "strokeWidth": 1.375,
                                            }
                                          }
                                        />
                                      </svg>
                                    </Component>
                                    <ScreenReaderOnlyText>
                                      <span
                                        style={
                                          Object {
                                            "border": 0,
                                            "clip": "rect(0,0,0,0)",
                                            "height": "1px",
                                            "margin": "-1px",
                                            "overflow": "hidden",
                                            "padding": 0,
                                            "position": "absolute",
                                            "width": "1px",
                                          }
                                        }
                                      >
                                        Close modal
                                      </span>
                                    </ScreenReaderOnlyText>
                                  </a>
                                </Transition>
                              </DefaultCloseButton>
                            </div>
                          </Column>
                        </div>
                      </Row>
                    </header>
                    <div
                      className="modal-body"
                    >
                      <div
                        style={
                          Object {
                            "position": "relative",
                          }
                        }
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                </BootstrapWrapper>
              </div>
            </div>
          </div>
        </FocusTrap>
      </Modal>
    </Portal>
  </Displaced>
</Modal>
`;

exports[`Modal nothing displayed if inactive 1`] = `""`;
